src/components/popover/widgets/capture-keystroke/capture-keystroke-button.component.ts

```diff
@@ -1,23 +1,77 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, EventEmitter, HostListener, Output } from '@angular/core';
+import { CaptureService } from '../../../../services/capture.service';
 
 @Component({
     selector: 'capture-keystroke-button',
     template: require('./capture-keystroke-button.component.html'),
     styles: [require('./capture-keystroke-button.component.scss')]
 })
-export class CaptureKeystrokeButtonComponent implements OnInit {
+export class CaptureKeystrokeButtonComponent {
+    @Output() capture = new EventEmitter<any>();
+
     private record: boolean;
+    private first: boolean;
+
+    constructor(private captureService: CaptureService) {
+        this.record = false;
+        this.captureService.initModifiers();
+        this.captureService.populateMapping();
+    }
 
-    constructor() { }
+    @HostListener('keyup')
+    onKeyUp() {
+        if (this.record && !this.first) {
+            this.saveScanCode();
+        }
+    }
+
+    @HostListener('keydown', ['$event'])
+    onKeyDown(e: KeyboardEvent) {
+        const code: number = e.keyCode;
 
-    ngOnInit() { }
+        if (this.record) {
+            this.first = false;
+
+            if (this.captureService.hasMap(code)) {
+                this.saveScanCode(this.captureService.getMap(code));
+            } else {
+                this.captureService.setModifier((e.location === 1), code);
+            }
+        } else if (code === 13) {
+            this.record = true;
+            this.first = true;
+        }
+    }
+
+    @HostListener('focusout')
+    onFocusOut() {
+        this.record = false;
+        this.reset();
+    }
 
     start(): void {
-        this.record = true;
+        if (!this.record) {
```

Comment:
Maybe we should talk this about in hangout, but I am strongly believe this function would behave the same as it was. 
It doesn't matter if there are other functions. I am talking about basic stuffs. 
Function has an input parameter this. 
if this.record is true then !this.record is false, so at the end of the function this.record is true.
if this.record is false then !this.record is true, then we set this.record to true, so at the end of the function this.record is true.
It doesn't matter who, when, why calls the start function after calling it this.record will be true.
So I would keep it as it was before.